lib/posix/timer.c

```diff
@@ -198,6 +257,18 @@ int timer_delete(timer_t timerid)
 		k_timer_stop(&timer->ztimer);
 	}
 
+	if (timer->sigev_notify == SIGEV_THREAD) {
+		pthread_cancel(timer->thread);
```

Comment:
should we `pthread_join` ?

Reply:
if thread takes long time to end, it become leads to hold here. but we need to stop the thread when timer is deleted right?
so thats why i use thread_cancel here